src/DynamoPackages/PackageLoader.cs

```diff
@@ -85,7 +86,28 @@ public string DefaultPackagesDirectory
             get { return defaultPackagesDirectoryIndex != -1 ? packagesDirectories[defaultPackagesDirectoryIndex] : null; }
         }
 
-        private int defaultPackagesDirectoryIndex = -1;
+        private static string TransformPath(string root, string userDataFolder, string extension)
+        {
+            if (root.StartsWith(userDataFolder))
+                return Path.Combine(root, extension);
+            try
+            {
+                var subFolder = Path.Combine(root, extension);
+                if (Directory.Exists(subFolder))
+                    return subFolder;
+            }
+            catch (IOException) { }
+            catch (ArgumentException) { }
+            catch (UnauthorizedAccessException) { }
+
+            return root;
+        }
+
+        internal void SetPackagesDownloadDirectory(string downloadDirectory, string userDataFolder)
+        {
+            defaultPackagesDirectoryIndex = packagesDirectories.IndexOf(
```

Comment:
what if this returns -1, is that acceptable?

Reply:
It was being assigned -1 even before, the core logic is the same as before so I didn't think much about it.